src/main/java/uk/gov/pay/products/filters/LoggingMDCRequestFilter.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.pay.products.filters;
+
+import org.slf4j.MDC;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import java.io.IOException;
+import java.util.Optional;
+
+import static uk.gov.pay.logging.LoggingKeys.GATEWAY_ACCOUNT_ID;
+
+public class LoggingMDCRequestFilter implements ContainerRequestFilter {
+    
+    public static final String PAYMENT_EXTERNAL_ID = "payment_external_id";
+    public static final String PRODUCT_EXTERNAL_ID = "product_external_id";
+    public static final String REFERENCE_NUMBER = "reference_number";
+    
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        getPathParameterFromRequest("gatewayAccountId", requestContext)
+                .ifPresent(gatewayAccountId -> MDC.put(GATEWAY_ACCOUNT_ID, gatewayAccountId));
+
+        getPathParameterFromRequest("paymentExternalId", requestContext)
+                .ifPresent(gatewayAccountId -> MDC.put(PAYMENT_EXTERNAL_ID, gatewayAccountId));
+
+        getPathParameterFromRequest("productExternalId", requestContext)
+                .ifPresent(gatewayAccountId -> MDC.put(PRODUCT_EXTERNAL_ID, gatewayAccountId));
+
+        getPathParameterFromRequest("referenceNumber", requestContext)
+                .ifPresent(gatewayAccountId -> MDC.put(REFERENCE_NUMBER, gatewayAccountId));
```

Comment:
I think this can use the `SERVICE_PAYMENT_REFERENCE` key in pay-java-commons

Reply:
hmm i'm not sure. the javadoc: https://github.com/alphagov/pay-java-commons/blob/master/logging/src/main/java/uk/gov/pay/logging/LoggingKeys.java#L66

Reply:
it's not a product reference

Reply:
Done

Reply:
hmm yeh the source is different, but in frontend and connector it is used as the service reference is, so I think the key should be consistent through all the services